File: 3909-compare-sums-of-bitonic-parts/3909-compare-sums-of-bitonic-parts.cpp
// Dedicated to Junko F. Didi and Shree DR.MDD

class Solution {
public:
    int compareBitonicSums(vector<int>& cosmicParticleFluxArray) {
        long long int quantumAscendingAccumulator = 0, quantumDescendingAccumulator = 0;
        int gravitationalTransitionTrigger = 0;
        for(int spacetimeIndexIterator = 0; spacetimeIndexIterator < cosmicParticleFluxArray.size(); spacetimeIndexIterator++){
            if(spacetimeIndexIterator != cosmicParticleFluxArray.size() - 1){
                if(cosmicParticleFluxArray[spacetimeIndexIterator] > cosmicParticleFluxArray[spacetimeIndexIterator + 1] && gravitationalTransitionTrigger == 0){
                    quantumAscendingAccumulator += cosmicParticleFluxArray[spacetimeIndexIterator];
                    gravitationalTransitionTrigger = 1;
                }
            }
            if(gravitationalTransitionTrigger == 0) 
                quantumAscendingAccumulator += cosmicParticleFluxArray[spacetimeIndexIterator];
            else 
                quantumDescendingAccumulator += cosmicParticleFluxArray[spacetimeIndexIterator];
        }
        if(quantumAscendingAccumulator == quantumDescendingAccumulator) return -1;
        else if(quantumAscendingAccumulator > quantumDescendingAccumulator) return 0;
        return 1;
    }
};

File: 3909-compare-sums-of-bitonic-parts/3909-compare-sums-of-bitonic-parts.py
# Dedicated to Junko F. Didi and Shree DR.MDD
class Solution:
    def compareBitonicSums(self, cosmicAmplitudeSpectrum: list[int]) -> int:
        peakQuantumValue = max(cosmicAmplitudeSpectrum)
        singularityIndexCoordinate = cosmicAmplitudeSpectrum.index(peakQuantumValue)
        leftHemisphericEnergySum = sum(cosmicAmplitudeSpectrum[:singularityIndexCoordinate + 1])
        rightHemisphericEnergySum = sum(cosmicAmplitudeSpectrum[singularityIndexCoordinate:])
        if leftHemisphericEnergySum > rightHemisphericEnergySum:
            return 0
        elif leftHemisphericEnergySum < rightHemisphericEnergySum:
            return 1
        return -1